src/proto/mod.rs

```diff
@@ -74,10 +74,12 @@ impl<T: Send + Sync + fmt::Debug> fmt::Debug for BroadcastMessage<T> {
 }
 
 /// Messages sent during the binary Byzantine agreement stage.
-#[derive(Copy, Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum AgreementMessage {
-    BVal(bool),
-    Aux(bool),
+    /// BVAL message with an epoch.
+    BVal((u32, bool)),
+    /// AUX message with an epoch.
+    Aux((u32, bool)),
```

Comment:
Maybe we should move the `AgreementMessage` type into the `agreement` module. I think the individual algorithms' modules could be pretty self-contained.